src/payment/strategies/zip/zip-payment-strategy.ts

```diff
@@ -74,7 +74,7 @@ export default class ZipPaymentStrategy implements PaymentStrategy {
         }
 
         if (this._redirectFlowIsTrue()) {
-            let nonce: { id: string };
+            let nonce: { id: string; uri: string };
             try {
                 nonce = JSON.parse(this._paymentMethod.clientToken);
```

Comment:
Why is the `uri` part of `clientToken` here? Also why is the API returning redirect_url as part of the client token?

Reply:
Hello, the same answer for this provider , We are using this approach due to that the zip/quadpay dev team has removed the `uri` data from the `Retrieve checkout` endpoint(the `get` call to the ` /checkouts/{Id}`), and the only way to get this redirect url is from the initialization call `POST` to `/checkouts`, so, we are passing this `uri` data through the Payment initialization call(where we are getting the `client_token`) to get the `uri` data and then redirect to the external zip/quadpay checkout.

Reply:
I understand why this is implemented. But this seems like a breaking change, since older versions of SDK will still expect it as part of the older contract.

Is there a way to change the implementation so as to it doesn't break older versions of SDK.